service-discovery-lagom13-java/src/main/java/com/lightbend/rp/servicediscovery/lagom/javadsl/LagomServiceLocator.java

```diff
@@ -41,17 +41,7 @@ public LagomServiceLocator(ActorSystem actorSystem, CircuitBreakers circuitBreak
     @Override
     public CompletionStage<Optional<URI>> locate(String name, Descriptor.Call<?, ?> serviceCall) {
         return com.lightbend.rp.servicediscovery.javadsl.ServiceLocator
-                .lookup(name, "lagom-api", actorSystem)
-                .thenApply(new Function<Optional<Service>, Optional<URI>>() {
-                    @Override
-                    public Optional<URI> apply(Optional<Service> service) {
-                        return service.map(new Function<Service, URI>() {
-                            @Override
-                            public URI apply(Service service) {
-                                return service.uri();
-                            }
-                        });
-                    }
-                });
+                .lookupOne(name, "lagom-api", actorSystem)
+                .thenApply(service -> service.map(Service::uri));
```

Comment:
Love it! I knew I could use lambda syntax here but didn't know the syntax :blush:, thanks for fixing this!

Reply:
You're welcome! 😄